src/Faithlife.Analyzers/EmptyStringAnalyzer.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Faithlife.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public sealed class EmptyStringAnalyzer : DiagnosticAnalyzer
+	{
+		public override void Initialize(AnalysisContext context)
+		{
+			context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+			{
+				var stringClass = compilationStartAnalysisContext.Compilation.GetTypeByMetadataName("System.String");
+				if (stringClass == null)
```

Comment:
Prefer `is null`. (I should create an analyzer!)

Reply:
Should we also prefer

```csharp
if (!(foo is null))
```

Reply:
For the reasons `foo is null` is better, yes. But because it's ugly, no. C# 8 pattern matching may allow `foo is {}` as an efficient pattern for "non-null object"; if so, we might as well wait until then to rewrite all the code. However, I'm completely unable to find this with a Google search or recall which GitHub issue discussed it. Which may indicate that it could be a bad idea to adopt a pattern that's primarily punctuation...

Reply:
https://twitter.com/jaredpar/status/999691652393414656

Not sure if there's been any significant development on this in the last year, though.

Reply:
Aha! I was thinking of this: https://twitter.com/MadsTorgersen/status/1053782748278677504

Doesn't work in latest SharpLab AFAICT though.

Reply:
Where do I vote for `foo aint null`?

Reply:
I think favoriting this tweet counts as a binding language design vote: https://twitter.com/Nick_Craver/status/999693192181186560

Reply:
https://twitter.com/jaredpar/status/1115019017297596416 recommends `foo is object` for non-null test. The only downside is that R# automatically warns for `IsExpressionAlwaysTrue` and that has to be disabled.